### ISSUE TITLE
communication between pods is too narrow.

### ISSUE THREAD

**Issue by kahirokunn** [terraform-aws-modules/terraform-aws-eks#1790]
## Description

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/9f85dc8cf5028ed2bab49f495f58e68aa870e7d4/node_groups.tf#L86

Communication with clusters in node_security_group_rules is fine, but communication between pods is too narrow.
For example, if you deploy an ArgoCD cluster, it will not work because the communication between pods does not work.
Since the communication between pods is basically limited by k8s, it is better to keep the default limitation as follows.
This is also how `eksctl` works.

```
{
  ingress_self_any = {
    description = "Node to node All"
    protocol    = "all"
    from_port   = 0
    to_port     = 65535
    type        = "ingress"
    self        = true
  }
  egress_self_any = {
    description = "Node to node All"
    protocol    = "all"
    from_port   = 0
    to_port     = 65535
    type        = "egress"
    self        = true
  }
}
```

**Comment by bryantbiggs** [comment]
this is by design in order to provide an extensible module that starts from a more secure standpoint. please see documentation for design considerations https://github.com/terraform-aws-modules/terraform-aws-eks#security-groups

it is left up to users to open up the additional network access as they see fit for their workloads (just like IAM permissions and security groups are no access by default, its up to users to add the necessary access)

**Comment by mjpowersjr** [comment]
I like the secure-by-default approach - but maybe a way to help new users would be to expand upon this part of the documentation w. additional examples on how to open up the networking?

**Comment by bryantbiggs** [comment]
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/9f85dc8cf5028ed2bab49f495f58e68aa870e7d4/examples/eks_managed_node_group/main.tf#L50

**Comment by kahirokunn** [comment]
Is there an example where traffic with the cluster is restricted, but node to node traffic is fully open?

**Comment by bryantbiggs** [comment]
@kahirokunn see #1805

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.2.3 :tada:
